Philips.CodeAnalysis.MaintainabilityAnalyzers/Naming/AvoidVariableNamedUnderscoreAnalyzer.cs

```diff
@@ -88,21 +110,94 @@ private void AnalyzeArgument(SyntaxNodeAnalysisContext context)
 				return;
 			}
 
-			// Check if it's a variable declaration
+			// Check if it's a typed discard (e.g., out int _)
 			if (argument.Expression is DeclarationExpressionSyntax declaration &&
-				declaration.Designation is SingleVariableDesignationSyntax variable)
+				declaration.Designation is DiscardDesignationSyntax discard)
 			{
-				if (variable.Identifier.ValueText != "_")
+				// This is a typed discard (e.g., out int _)
+				// Only flag if the type is not needed for overload resolution
+				if (!IsTypedDiscardNecessaryForOverloadResolution(context, argument))
 				{
-					return;
+					Location location = discard.GetLocation();
+					var diagnostic = Diagnostic.Create(UnnecessaryTypedDiscardRule, location, "_");
+					context.ReportDiagnostic(diagnostic);
 				}
+				return;
+			}
 
+			// Check if it's a variable declaration with identifier "_"
+			if (argument.Expression is DeclarationExpressionSyntax declaration2 &&
+				declaration2.Designation is SingleVariableDesignationSyntax variable &&
+				variable.Identifier.ValueText == "_")
+			{
 				Location location = variable.Identifier.GetLocation();
-				var diagnostic = Diagnostic.Create(Rule, location, variable.Identifier.ValueText);
+				var diagnostic = Diagnostic.Create(VariableNamedUnderscoreRule, location, variable.Identifier.ValueText);
 				context.ReportDiagnostic(diagnostic);
-				// Note: DiscardDesignationSyntax represents proper discards (e.g., "out _" or "out string _")
-				// and should not be flagged, so we don't handle this case.
 			}
+
+			// Note: We don't flag IdentifierNameSyntax("_") because that represents
+			// anonymous discards like "out _" which are the preferred form
+		}
+
+		private static bool IsTypedDiscardNecessaryForOverloadResolution(SyntaxNodeAnalysisContext context, ArgumentSyntax argument)
+		{
+			// Find the method call containing this argument
+			InvocationExpressionSyntax invocation = argument.Ancestors()
+				.OfType<InvocationExpressionSyntax>()
+				.FirstOrDefault();
+			if (invocation is null)
+			{
+				return false; // Can't find method call, allow the flag
+			}
+
+			// Get semantic information about the method
+			SymbolInfo symbolInfo = context.SemanticModel.GetSymbolInfo(invocation);
```

Comment:
Put the `SemanticModel` as low as possible in this method, for performance reasons.

Reply:
In fact, maybe you can do without the `SemanticModel`